docs/async.md

```diff
@@ -0,0 +1,210 @@
+client = httpx.AsyncClient(backend='asyncio')  # Use asyncio as the backend.
+client = httpx.AsyncClient(backend='trio')     # Use trio as the backend.
+```
+
+### [AsyncIO](https://docs.python.org/3/library/asyncio.html)
+
+AsyncIO is Python's [built-in library](https://docs.python.org/3/library/asyncio.html)
+for writing concurrent code with the async/await syntax.
+
+```python
+import asyncio
+import httpx
+
+async def main():
+    client = httpx.AsyncClient()
+    response = await client.get('https://www.example.com/')
+    print(response)
+
+asyncio.run(main())
+```
+
+### [Trio](https://github.com/python-trio/trio)
+
+Trio is [an alternative async library](https://trio.readthedocs.io/en/stable/),
+designed around the [the principles of structured concurrency](https://en.wikipedia.org/wiki/Structured_concurrency).
+
+```python
+import httpx
+import trio
+
+async def main():
+    client = httpx.AsyncClient()
+    response = await client.get('https://www.example.com/')
+    print(response)
+
+trio.run(main)
+```
+
+!!! important
+    The `trio` package must be installed to use the Trio backend.
+
+## Calling into Python Web Apps
+
+Just as `httpx.Client` allows you to call directly into WSGI web applications,
+the `httpx.AsyncClient` class allows you to call directly into ASGI web applications.
+
+Let's take this Starlette application as an example:
+
+```python
+from starlette.applications import Starlette
+from starlette.responses import HTMLResponse
+from starlette.routing import Route
+
+
+async def hello():
+    return HTMLResponse("Hello World!")
+
+
+app = Starlette(routes=[Route("/", hello)])
+```
+
+We can make requests directly against the application, like so:
+
+```python
+>>> import httpx
+>>> async with httpx.AsyncClient(app=app) as client:
+...     r = await client.get('http://example/')
+...     assert r.status_code == 200
+...     assert r.text == "Hello World!"
+```
+
+For some more complex cases you might need to customise the ASGI dispatch. This allows you to:
+
+* Inspect 500 error responses rather than raise exceptions by setting `raise_app_exceptions=False`.
+* Mount the ASGI application at a subpath by setting `root_path`.
+* Use a given client address for requests by setting `client`.
+
+For example:
+
+```python
+# Instantiate a client that makes ASGI requests with a client IP of "1.2.3.4",
+# on port 123.
+dispatch = httpx.dispatch.ASGIDispatch(app=app, client=("1.2.3.4", 123))
+async with httpx.AsyncClient(dispatch=dispatch) as client:
+    ...
+```
+
+See [the ASGI documentation](https://asgi.readthedocs.io/en/latest/specs/www.html#connection-scope) for more details on the `client` and `root_path` keys.
+
+## Unix Domain Sockets
+
+The async client provides support for connect through a unix domain socket via the `uds` parameter. This is useful when making requests to a server that is bound to a socket file rather than an IP address.
```

Comment:
```suggestion
The async client provides support for connecting through a unix domain socket via the `uds` parameter. This is useful when making requests to a server that is bound to a socket file rather than an IP address.
```